### ISSUE TITLE
[feat] RH-Table supports dark theme 

### ISSUE THREAD

**Issue by LyndseyR** [RedHat-UX/red-hat-design-system#1371]
### Description

Be able to add an rh-table to a dark background 

### Suggested solution

Have text inside rh-table show white or light gray on a dark background

### Additional context

Tables added in the WYSIWYG reference rh-table and right now it only works on white/light-gray backgrounds 

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/20518754/9d3d216f-096e-4dcb-9723-3931658554ee)

To test

1. add a section on a RHDC page with a black background and set theme to dark
2. bring in simple text component 
3. inside WYSIWYG add a table 
4. Save the block
5. Text is now black on black

**Comment by markcaron** [comment]
@zeroedin how are we doing on this?

@marionnegp will provide a mockup for the dark theme and images for the docs.

**Comment by marionnegp** [comment]
@coreyvickery, I created a [separate file](https://www.figma.com/file/WPYBURa3ln1FBcjhLx5xds/Element---table---style-page---dark-theme-update?type=design&node-id=0%3A1&mode=design&t=rdciYzf5o9YexfzJ-1) for the `<rh-table>` Style page updates. Let me know if these look good to you. I've also invited you to edit the file.

Changes:
- Added a "Theme" section
- Added dark theme hover and focus mockups

Here's a [dev mode link](https://www.figma.com/file/WPYBURa3ln1FBcjhLx5xds/Element---table---style-page---dark-theme-update?type=design&node-id=0%3A1&mode=dev&t=rdciYzf5o9YexfzJ-1) to the same file, which can be used for both the element and docs updates.

**Comment by zeroedin** [comment]
> @zeroedin how are we doing on this?

PR should be linked to this issue.  Got as far as I could just waiting on colors from design.

Here is the demo in the DP: https://deploy-preview-1373--red-hat-design-system.netlify.app/elements/table/demo/color-context/

**Comment by zeroedin** [comment]
> Here's a dev mode link to the same file, which can be used for both the element and docs updates.

I'll take a look at this as soon as possible and update my PR where needed.

**Comment by zeroedin** [comment]
@marionnegp looks like the figma file is taking the approach of `rh-accordion` where only `darkest` and `lightest` supported?  

If this is not the case I'd need colors for each variation as described in the DP demo above.  The DP color values are just ad hoc and haven't been even tested for accessibility so doing so would add that additional complexity.

If we are only supporting `darkest` `lightest` as internal contexts then could you provide examples of what this would look like if set in a `<rh-surface color-palette="dark">`. 

An example would be accordion providing its own internal `darkest` background 

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/d801beb6-91fc-4b0c-90be-acb6add0c9ad)

**Comment by marionnegp** [comment]
@zeroedin, thanks for calling that out. I changed the hover state's row bg color to `--rh-color-surface-dark-alt` at 70% opacity. That should show up on our three dark theme surface colors. 

Is `<rh-table>` currently a context provider? I don't think it should be. Those cells should have a no background color unless they're in the hover state.

**Comment by marionnegp** [comment]
> Is `<rh-table>` currently a context provider? I don't think it should be. Those cells should have a no background color unless they're in the hover state.

@coreyvickery, since you designed `<rh-table>`, do you think it would need to be context provider?

**Comment by zeroedin** [comment]
> changed the hover state's row bg color to `--rh-color-surface-dark-alt` at 80%

Surface tokens dont come with RGB and opacity, going to default to the crayon value for the gray-80.

```
  var(--rh-color-gray-80-rgb, rgb(41, 41, 41) / var(--rh-opacity-80, 80%));
```

Maybe we need to spin this off as a separate issue.

**Comment by zeroedin** [comment]
> Is `<rh-table>` currently a context provider? I don't think it should be.

I actually removed the context-provider decorator, as I also didn't really think it needed to be there.  

If we are moving forward with `rh-surface` as a parent element to provide the color-palette, then the table itself doesn't need to set a specific state on its own as its only ever reactive to that parent context.  Also table cells shouldn't be populated with other elements that need context, and even if they where they wouldn't need it from table specifically, they would be updated by the parent context as well.

An argument could be made that `rh-table` could exist on its own in a framed application or something, however that also could easily just be said that you'd still wrap it with a `rh-surface` even if it is alone on a region of a page to add the color palette contexts.

Another argument could be made that providing that attribute it is easier to key off of for lightdom css to decrease FOUC.  But given a requirement for the `rh-surface` as a parent I can just write a descendant selector that does the same thing, which is what i did: `[color-palette^="dark"] rh-table table { ... }`

**Comment by marionnegp** [comment]
@coreyvickery @zeroedin, using the light blue (`#92c5f9`) for the dark theme hover color didn't work for cells that have linked text unfortunately.
<img width="702" alt="Screenshot 2023-12-15 at 4 25 21 PM" src="https://github.com/RedHat-UX/red-hat-design-system/assets/95588923/fa51536c-c75f-4922-a4a8-9e6b0356fa3c">

I tried other combinations of white, grays, and our two interactive dark theme tokens, but using white for the row and `#003366` for the column provided enough contrast against all surface colors. You can check that out in the [Figma file](https://www.figma.com/file/WPYBURa3ln1FBcjhLx5xds/Element---Table---style-page---dark-theme-update?type=design&node-id=10%3A373&mode=dev&t=Vc2AlTC55kEns61k-1). I left some swatches of the cell's background color if you want to check contrast ratios. Corey, feel free to edit the file.
